### PR TITLE
TST: Mark coordinates tests as requiring remote data

### PR DESCRIPTION
**Pull request by pllim**
As part of #7928 but more controversial than #7989. This one is to address the following warnings I get when I run `python setup.py test -P coordinates` (without `--remote-data`).

```
An attempt was made to connect to the internet by a test that was not marked `remote_data`.
The requested host was: maia.usno.navy.mil
```

I looked at one of the tests and traced that one to such a call that invokes IERS, which in turns invokes Internet connection.

```
astropy/coordinates/builtin_frames/utils.py:40: in get_polar_motion
    xp, yp, status = iers.IERS_Auto.open().pm_xy(time, return_status=True)
```

I am not sure if this is a proper way to address such warnings but at least it is how I could get rid of them with minimal change. With this patch, here is a log of `python setup.py test -P coordinates` showing that the tests are now skipped without `--remote-data`: 
[ztstlog.txt](https://github.com/astropy/astropy/files/2528379/ztstlog.txt)

### CODE REVIEW

**Comment by pllim** [comment]
This is not release critical, re-milestoning.

**Comment by bsipocz** [comment]
@pllim - can you rebase? I need to do a new release soonish and can't see a reason why this should be kept open rather than merged in.

**Comment by pllim** [comment]
@bsipocz , it is rebased but now I wonder how much of it was a symptom that has since been fixed by #8163 . I don't have time to look into this now. Maybe we should not rush this one and it is not critical for release. Thanks!

**Comment by pllim** [comment]
@bsipocz 

UPDATE: I reran the tests with and without this patch and confirmed that this is **still necessary** to silence those warnings. In fact, I caught one more (probably it was added since I originally ran this). I think this is good to merge if you don't see anything wrong with it.

File: astropy/coordinates/tests/test_iau_fullstack.py [hunk 1]
```diff
@@ -29,7 +30,10 @@ def fullstack_icrs():
 def fullstack_fiducial_altaz(fullstack_icrs):
     altazframe = AltAz(location=EarthLocation(lat=0*u.deg, lon=0*u.deg, height=0*u.m),
                        obstime=Time('J2000'))
-    return fullstack_icrs.transform_to(altazframe)
+    with warnings.catch_warnings():  # Ignore remote_data warning
```
Comment (pllim) [inline astropy/coordinates/tests/test_iau_fullstack.py]: My fixture-fu is not great, so if there is a better way to do this, please let me know.

File: astropy/coordinates/tests/accuracy/test_altaz_icrs.py [hunk 2]
```diff
@@ -149,7 +152,8 @@ def test_against_jpl_horizons():
     assert distance < 1 * u.arcsec
 
 
-@pytest.mark.xfail
+@pytest.mark.remote_data
+@pytest.mark.xfail(reason="Current output is completely incorrect")
```
Comment (bsipocz) [inline astropy/coordinates/tests/accuracy/test_altaz_icrs.py]: ouch. any chance to fix this @adrn/@eteq? I mean in a new PR not in this one.

**Comment by pllim** [comment]
p.s. I am not convinced that this needs to be backported all the way to 2.x. Please remilestone to something less painful if straight backport is not possible.

File: astropy/coordinates/tests/accuracy/test_altaz_icrs.py (hunk 2 above)
Comment (adrn) [reply astropy/coordinates/tests/accuracy/test_altaz_icrs.py]: I made an issue to remind us #8296

**Comment by pllim** [comment]
Looks like this is now needed because of IERS being down and is holding up other PRs because the test is running when it is not supposed to without this patch. Merging.

**Pull request merged**